evap/staff/templates/staff_course_comments_quick.html

```diff
@@ -0,0 +1,436 @@
+                        this.index--;
+                    } else if(direction === "right" && this.isReviewing) {
+                        this.index++;
+                    }
+
+                    this.updateState(noFooter);
+                    this.answers.slide(this.active, direction);
+                },
+
+                press: function(action, toggle) {
+                    if(!this.isReviewing || action === "make_private" && !this.active.data("contribution")) {
+                        return;
+                    }
+
+                    if(toggle && this.active.attr("data-review") === action) {
+                        action = "unreview";
+                    }
+
+                    let parameters = {"id": this.active.data("id"), "action": action, "course_id": {{ course.id }}};
+                    $.ajax({
+                        type: "POST",
+                        url: "{% url 'staff:course_comments_update_publish' %}",
+                        data: parameters,
+                        error: function(){ window.alert("{% trans 'The server is not responding.' %}"); }
+                    });
+
+                    if(action === "unreview") {
+                        this.active.removeAttr("data-review");
+                        this.updateState();
+                        return;
+                    }
+
+                    this.active.attr("data-review", action);
+                    this.updateState();
+
+                    this.slide("right", true);
+                    this.focusedButton = this.slider.find("[data-action=" + action + "]");
+                    this.focusedButton.focus();
+                    clearTimeout(this.buttonTimeout);
+                    this.buttonTimeout = setTimeout($.proxy(this.finishSlide, this), 200);
+                },
+
+                finishSlide: function() {
+                    this.updateState();
+                    this.focusedButton.blur();
+                },
+
+                repeat: function(action) {
+                    let reviewed = [];
+                    let unreviewed = [];
+
+                    this.answers.items.each(function() {
+                        let item = $(this);
+                        if(item.is("[data-review]")) {
+                            reviewed.push(item);
+                        } else {
+                            unreviewed.push(item);
+                        }
+                    });
+
+                    if(action !== "all" && unreviewed.length) {
+                        this.index = reviewed.length;
+                    } else {
+                        this.index = 0;
+                    }
+
+                    this.items = reviewed.concat(unreviewed);
+
+                    let direction = action === "start"? "right" : "left";
+                    this.answers.slide(this.active, direction);
+                    this.updateState();
+                },
+
+                updateState: function(noFooter) {
```

Comment:
updateState seems to be a catch-all "update all the things", aren't there separate parts (e.g. the arrows and the buttons) that don't always need to be updated at the same time?